### DISCUSSION TITLE
bug - svg className in handleOnCellClick

### DISCUSSION THREAD

**Discussion by lkwheel** [ghiscoding/Angular-Slickgrid#846] [General]
The handleOnCellClick() function throws an error if the targetElm is an SVG. 
```
    // --
    // private functions
    // ------------------
    handleOnCellClick(event, args) {
        var _a;
        if (event && args) {
            const targetElm = event.target || {};
            const idPropName = (_a = this.gridOptions.datasetIdPropertyName) !== null && _a !== void 0 ? _a : 'id';
            const collapsedPropName = this.getTreeDataOptionPropName('collapsedPropName');
            const childrenPropName = this.getTreeDataOptionPropName('childrenPropName');
            if (targetElm === null || targetElm === void 0 ? void 0 : targetElm.className) {
                const hasToggleClass = targetElm.className.indexOf('toggle') >= 0 || false;
                if (hasToggleClass) {
                    const item = this.dataView.getItem(args.row);
```
I believe this is due to the indexOf use on the className. SVGs have an SVGAnimatedString object as its className and not a string as is typically the case.

**Comment by ghiscoding** [comment]
is that Angular-Slickgrid 3.x? If so then the code is [here](https://github.com/ghiscoding/slickgrid-universal/blob/master/packages/common/src/services/gridEvent.service.ts#L91), Pull Request of a fix is welcome. If however it's 2.x, then sorry but I stopped supporting it. This is an Open Source project feel free to contribute, I never use it with SVG.

Maybe another option is to use your own code via the `onClick` event directly from the grid, the code I wrote in the [gridEvent.service.ts](https://github.com/ghiscoding/slickgrid-universal/blob/master/packages/common/src/services/gridEvent.service.ts) is just a wrapper of the `onClick` event to make it easier to write code directly in the column definition instead of a global event but it's still an option.
